src/test/java/reactor/pool/CommonPoolTest.java

```diff
@@ -2744,4 +2745,70 @@ void releaseHandlerFailsThenDestroyHandlerAlsoFails(PoolStyle style) {
 		assert ref3 != null;
 		assertThat(ref3.poolable()).as("poolable 3").isEqualTo("bothReleaseHandlerAndDestroyHandlerFail3");
 	}
+
+	@ParameterizedTestWithName
+	@EnumSource
+	void invalidateIgnoredAfterReleasedDoesntAffectMetrics(PoolStyle style) {
+		AtomicInteger releaseCount = new AtomicInteger();
+		AtomicInteger destroyCount = new AtomicInteger();
+		PoolBuilder<String, PoolConfig<String>> configBuilder = PoolBuilder
+				.from(Mono.fromCallable(() -> "invalidateIgnoredAfterReleased"))
+				.sizeBetween(0, 1)
+				.releaseHandler(it -> Mono.fromRunnable(releaseCount::incrementAndGet))
+				.destroyHandler(it -> Mono.fromRunnable(destroyCount::incrementAndGet));
+
+		InstrumentedPool<String> pool = style.apply(configBuilder);
+
+		PooledRef<String> ref1 = pool.acquire().block();
+		assert ref1 != null;
+		assertThat(pool.metrics().acquiredSize()).as("acquired metric pre-release").isOne();
+		assertThat(pool.metrics().allocatedSize()).as("allocated metric pre-release").isOne();
+		assertThat(pool.metrics().idleSize()).as("idle metric pre-release").isZero();
+		assertThat(releaseCount).as("released pre-release").hasValue(0);
+		assertThat(destroyCount).as("destroyed pre-release").hasValue(0);
+
+		ref1.release().block();
```

Comment:
What will happen if the first call is `invalidate` and the second call is `release`?

Reply:
the resource would be destroyed (`destroyCount == 1`) and all 3 metrics `acquiredSize()`/`allocatedSize()`/`idleSize()` would be at zero.

I have added a test to show that